nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -256,6 +419,311 @@ impl DataStore {
             })
     }
 
+    pub async fn ip_pool_association_list(
+        &self,
+        opctx: &OpContext,
+        authz_pool: &authz::IpPool,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<IpPoolResource> {
+        use db::schema::ip_pool;
+        use db::schema::ip_pool_resource;
+
+        paginated(
+            ip_pool_resource::table,
+            ip_pool_resource::ip_pool_id,
+            pagparams,
+        )
+        .inner_join(ip_pool::table)
+        .filter(ip_pool::id.eq(authz_pool.id()))
+        .filter(ip_pool::time_deleted.is_null())
+        .select(IpPoolResource::as_select())
+        .load_async::<IpPoolResource>(
+            &*self.pool_connection_authorized(opctx).await?,
+        )
+        .await
+        .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+    }
+
+    // TODO: should this error on conflict instead of updating?
+    pub async fn ip_pool_associate_resource(
```

Comment:
Minor nit, but should we use `link` here?

Reply:
Yes. And it should really be called `link_silo`